R/plots.R

```diff
@@ -410,7 +410,43 @@ plot_3d_animated <- function(data, grid_coords, start, end, output_dt,
 #' }
 single_emission_rate_plot <- function(sim_dt, puff_dt, output_dt, start_time, end_time,
                                       source_coords, emission_rate, wind_data, sensor_coords) {
-  # Function implementation
+  sensor_concentrations <- simulate_sensor_mode(

```

Comment:
Please cut from here through line 423. No need to fit the model within the plotting function as that would be pretty inefficient and potential very slow. Rather, in line with the spirit of the other plotting functions, keep required inputs as minimal as possible, and likely only a list object or the output from running `simulate_sensor_mode()` elsewhere.